appwindow.js

```diff
@@ -107,6 +107,24 @@ var AppWindow = GObject.registerClass(
             this.method_handler(this.settings, 'changed::window-position', this.update_resize_boxes);
             this.update_resize_boxes();
 
+            const HEIGHT_MOD = 0.05;
+            this.simple_action('window-size-dec', () => {
+                if (this.settings.get_boolean('window-maximize'))
+                    this.settings.set_double('window-size', 1.0);
+                this.adjust_double_setting('window-size', -HEIGHT_MOD);
+            });
+            this.simple_action('window-size-inc', () => {
+                this.adjust_double_setting('window-size', HEIGHT_MOD);
```

Comment:
This too doesn't work correctly if `window-maximize` is `true` and `window-size` < `0.95`. Simply doing nothing if `window-maximize` is `true` should be fine